ledger/participant-integration-api/src/main/scala/platform/apiserver/services/ApiMeteringReportService.scala

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.apiserver.services
+
+import com.daml.ledger.api.v1.admin.metering_report_service.MeteringReportServiceGrpc.MeteringReportService
+import com.daml.ledger.api.v1.admin.metering_report_service._
+import com.daml.logging.{ContextualizedLogger, LoggingContext}
+import com.daml.platform.api.grpc.GrpcApiService
+import io.grpc.ServerServiceDefinition
+
+import scala.concurrent.{ExecutionContext, Future}
+
+private[apiserver] final class ApiMeteringReportService()(implicit
+    executionContext: ExecutionContext,
+    loggingContext: LoggingContext,
+) extends MeteringReportService
+    with GrpcApiService {
+
+  private implicit val logger: ContextualizedLogger = ContextualizedLogger.get(this.getClass)
+
+  override def bindService(): ServerServiceDefinition =
+    MeteringReportServiceGrpc.bindService(this, executionContext)
+
+  override def close(): Unit = ()
+
+  override def getMeteringReport(
+      request: GetMeteringReportRequest
+  ): Future[GetMeteringReportResponse] = {
+    logger.info(s"Received metering report request: $request")
+    val generationTime =
+      com.google.protobuf.timestamp.Timestamp.of(java.time.Instant.now().getEpochSecond, 0)
```

Comment:
You are using a protobuf Timestamp, which stores a nanosecond part. For uniformity of our codebase, I'd suggest to store the nanoseconds as well, given that `java.time.Instant` provides them.

Reply:
Done.